src/command.rs

```diff
@@ -271,226 +305,351 @@ pub fn parse(input: &str) -> Option<Vec<Command>> {
-                        match first_char {
-                            // handle i32::MAX < unsigned_millis < u32::MAX gracefully
-                            Some('+') => {
-                                i32::try_from(unsigned_millis)
-                                    .ok()
-                                    .map(|unsigned_millis_i32| {
-                                        SeekDirection::Relative(unsigned_millis_i32)
-                                    })
-                            }
-                            Some('-') => {
-                                i32::try_from(unsigned_millis)
-                                    .ok()
-                                    .map(|unsigned_millis_i32| {
-                                        SeekDirection::Relative(-unsigned_millis_i32)
-                                    })
-                            }
-                            _ => Some(SeekDirection::Absolute(unsigned_millis)),
-                        }
-                        .map(|direction| Command::Seek(direction))
-                    })
+                let unsigned_millis = match duration_raw.parse() {
+                    // accept raw milliseconds
+                    Ok(millis) => millis,
+                    Err(_) => parse_duration::parse(&duration_raw) // accept fancy duration
+                        .map_err(|err| ArgParseError {
+                            arg: duration_raw.clone(),
+                            err: err.to_string(),
+                        })
+                        .and_then(|dur| {
+                            dur.as_millis().try_into().map_err(|_| ArgParseError {
+                                arg: duration_raw.clone(),
+                                err: "Duration value too large".into(),
+                            })
+                        })?,
+                };
+                let seek_direction = match first_char {
+                    // handle i32::MAX < unsigned_millis < u32::MAX gracefully
+                    Some('+') => {
+                        i32::try_from(unsigned_millis).map(|millis| SeekDirection::Relative(millis))
+                    }
+                    Some('-') => i32::try_from(unsigned_millis)
+                        .map(|millis| SeekDirection::Relative(-millis)),
+                    _ => Ok(SeekDirection::Absolute(unsigned_millis)),
+                }
+                .map_err(|_| ArgParseError {
+                    arg: duration_raw,
+                    err: "Duration value too large".into(),
+                })?;
+                Command::Seek(seek_direction)
+            }
+            "focus" => {
+                let &target = args.get(0).ok_or(InsufficientArgs {
+                    cmd: command.into(),
+                    hint: Some("queue|search|library".into()),
+                })?;
+                // TODO: this really should be strongly typed
+                Command::Focus(target.into())
```

Comment:
`Command::Focus` currently accepts a `String`. It is possible to strongly type it with an enum?

Reply:
I have no preference here, both are fine to me. I think the gain is rather low compared to the effort, as `Layout` already checks if the screen exists when processing the command, but feel free to type it if you prefer.

Reply:
You're right. It's probably more effort than it's worth unless we're actually considering adding another screen. This can wait for when (and if) that situation arises.